x-pack/plugins/monitoring/server/lib/apm/get_apm_info.js

```diff
@@ -0,0 +1,139 @@
+  };
+}
+
+export async function getApmInfo(req, apmIndexPattern, { clusterUuid, apmUuid, start, end }) {
+  checkParam(apmIndexPattern, 'apmIndexPattern in beats/getBeatSummary');
+
+  const filters = [
+    { term: { 'beats_stats.beat.uuid': apmUuid } },
+    { term: { 'beats_stats.beat.type': 'apm-server' } }
+  ];
+  const params = {
+    index: apmIndexPattern,
+    size: 1,
+    ignoreUnavailable: true,
+    filterPath: [
+      'hits.hits._source.beats_stats.beat.host',
+      'hits.hits._source.beats_stats.beat.version',
+      'hits.hits._source.beats_stats.beat.name',
+      'hits.hits._source.beats_stats.beat.type',
+      'hits.hits._source.beats_stats.metrics.libbeat.output.type',
+      'hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.published',
+      'hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.total',
+      'hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.dropped',
+      'hits.hits._source.beats_stats.metrics.libbeat.output.write.bytes',
+      'hits.hits._source.beats_stats.metrics.libbeat.config.reloads',
+      'hits.hits._source.beats_stats.metrics.beat.info.uptime.ms',
+      'hits.hits.inner_hits.first_hit.hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.published',
+      'hits.hits.inner_hits.first_hit.hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.total',
+      'hits.hits.inner_hits.first_hit.hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.dropped',
+      'hits.hits.inner_hits.first_hit.hits.hits._source.beats_stats.metrics.libbeat.output.write.bytes',
+    ],
+    body: {
+      sort: { timestamp: { order: 'desc' } },
+      query: createQuery({
+        start,
+        end,
+        clusterUuid,
+        metric: ApmMetric.getMetricFields(),
+        filters
+      }),
+      collapse: {
+        field: 'beats_stats.metrics.beat.info.ephemeral_id', // collapse on ephemeral_id to handle restart
+        inner_hits: {
+          name: 'first_hit',
+          size: 1,
+          sort: { 'beats_stats.timestamp': 'asc' }
+        }
+      }
+    }
+  };
+
+  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
+
+  // Fetch additional information
+  const params2 = {
+    index: apmIndexPattern,
+    size: 0,
+    ignoreUnavailable: true,
+    // filterPath: apmAggFilterPath,
+    body: {
+      query: createQuery({
+        start,
+        end,
+        clusterUuid,
+        metric: ApmMetric.getMetricFields(),
+        filters
+      }),
+      aggs: {
+        acked_events: {
+          filter: {
+            range: {
+              'beats_stats.metrics.libbeat.output.events.acked': {
+                gt: 0,
+              }
+            }
+          },
+          aggs: {
+            max_timestamp: {
+              max: {
+                field: 'timestamp'
+              }
+            }
+          }
+        }
+      }
+    }
+  };
+
+  const [response, response2] = await Promise.all([
```

Comment:
Same for `response2` as for `params2`. If we have descriptive variable names I think it should make it much easier for us getting back to the code or if someone else has to touch it.